common/consistenthash/consistenthash.go

```diff
@@ -22,7 +23,22 @@ func (r *Ring) search(key string) int {
     // YOUR CODE GOES HERE //
     /////////////////////////
 
-    return 0
+	// Verificando a soma hash da chave
+    hashSum := hashId(key)
+    index := 0
+
+	// Percorrendo o anel no sentido horário até que seja encontrada a posição correta (nó correto)
+	// Após encontrar a posição correta, pare!
+	log.Printf("[SEARCH] Searching ring node clockwise. Please, wait.\n")
+    for nodePos, node := range r.Nodes {
+		log.Printf("[SEARCH] Hostname = %s - hashId = %d - Ring Node = %d.\n", node.Id, node.HashId, nodePos)
+        index = nodePos
+        if hashSum <= node.HashId {
```

Comment:
Se o hash da chave dada for maior que todos os hashs no anel, o nó responsável por ela é o de índice 0 (dá a volta no anel). No seu código, seria retornado o último valor processado aqui.